src/status_im/ui2/screens/chat/photo_selector/view.cljs

```diff
@@ -78,14 +54,21 @@
    (when (some #{item} @selected)
      [info-count/info-count (+ (utils/first-index #(= item %) @selected) 1) (style/image-count)])])
 
+(defn on-end-reached [end-cursor]
```

Comment:
@OmarBasem, I really think this kind of logic should be in event handlers. So you can define an event, and in the `:on-end-reached` of the flat list you can dispatch it. For an example, see `src/status_im/ui/screens/activity_center/views.cljs`, function `activity center`.

Reply:
I think that's what I am doing. The difference is that here we are dispatching based on a condition first.

Reply:
What I mean is that this function shouldn't be part of UI namespaces, does it make sense? It seems to me the function should be defined as an event handler instead.

Reply:
@ilmotta I think it is better to keep it that way where it checks for conditions before dispatching. Otherwise, it would be unnecessarily dispatching events (in a spamful way) every time it reaches `end`

Reply:
I disagree @OmarBasem, I think this is real misconception in the code.

1. Subscriptions have a cost and they should ideally be left for UI components according to re-frame. `on-end-reached` is not a UI component. In fact, it's just a function that needs to read some state and dispatch two effects. This is exactly what an event handler is supposed to do.

2. The `on-end-reached` function will need to be called by re-frame every time either `camera-roll/loading-more` or `camera-roll/has-next-page` state changes and then further dispatch. Every function that subscribes to something will need to be potentially called multiple times due to rendering cycles. This is more work than only dispatching if the RN FlatList detects the onEndReachedThreshold has been reached.

3. If we follow this very logic we would then have to move lots and lots of event handlers to the view layer and rely on needless subscriptions. This would dramatically hurt the performance of the app and break re-frame's recommendations.

Reply:
Thanks @ilmotta for the explanation. Fixed it.